### ISSUE TITLE
Let the gm track other per PC things in the app

### ISSUE THREAD

**Issue by devonjones** [cynicaloptimist/improved-initiative#41]
It would be cool if there was an ability to just add boxes on a character, say like it's own field where you could track checkboxes that can be reset by a long or short rest.

This would imply that there would also be a "long rest" and "short rest" button that resets these checkboxes.

it would be kinda helpful to track hit dice used in a short rest this way too.

**Comment by cynicaloptimist** [comment]
Adding some background for the interested, and to work through some details:

As of this writing, the app has two separate concepts, a `StatBlock` and a `Combatant`. The `StatBlock` represents the content that you would find on the page in the Monster Manual, while the Combatant represents the notes you'd take about the creature during a combat (such as tracking HP and initiative). When a `Combatant` is added to an `Encounter`, a copy of its `StatBlock` is attached to it. This means you can edit the `StatBlock` on that `Combatant` without it affecting the original copy of the `StatBlock` in your library. This works fine for creatures that will be used in a single `Encounter`.

Simply put, Player Characters are just special `Combatant`s that reveal their HP in the player view, and are saved in a different library. There are a few other ways that they differ, but that's the main idea.

What may be missing from this model is a separate, persistent store of information that refers to a unique instance of a `StatBlock` but exists longer than a single `Encounter` like `Combatant` does today.

**Comment by AJAnderson** [comment]
@cynicaloptimist I think your assessment of what's required is spot on. I was a little surprised when I first used the app that you couldn't persist the player state from encounter to encounter. I'm not sure how that would be implemented practically. Perhaps by fundamentally changing what a "player" is, like you say.
